kenya_compliance/kenya_compliance/overrides/server/stock_ledger_entry.py

```diff
@@ -78,9 +87,7 @@ def on_update(doc: Document, method: str | None = None) -> None:
 
     if doc.voucher_type == "Stock Entry":
         items_list = get_stock_entry_movement_items_details(record.items, all_items)
-        current_item = list(
-            filter(lambda item: item["itemNm"] == doc.item_code, items_list)
-        )
+        current_item = [item for item in items_list if item["itemNm"] in [i.item_code for i in doc.items]]
```

Comment:
You've replaced the loop with one that runs in Quadratic time. This operation will be less efficient